ts/src/gameClasses/EntityTrack.ts

```diff
@@ -0,0 +1,174 @@
+class EntityTrack {
+	trackEntityById: {[key: string]: IgeEntity};
+	timeStamp: number;
+
+	constructor() {
+		console.log('EntityTrack born');
+
+		this.trackEntityById = {};
+	}
+
+	applyRendererEvents(): void {
+		ige.client.on('tick', this.frameTick, this);
+	}
+
+	updateAllEntities (/*timeStamp*/): void {
+		var currentTime = Date.now();
+		if (!ige.lastTickTime) ige.lastTickTime = currentTime;
+		var tickDelta = currentTime - ige.lastTickTime;
+
+		// var entityCount = {unit: 0, item:0, player:0, wall:0, projectile: 0, undefined: 0, floatingLabel: 0}
+		for (var entityId in this.trackEntityById) {
+			//need to check if we need it, seems it is not affecting game
+			/*if (this.trackEntityById[entityId]._pixiContainer && this.trackEntityById[entityId]._pixiContainer._destroyed) {
+				delete this.trackEntityById[entityId];
+				break;
+			}*/
+
+			var entity = ige.$(entityId);
+			if (entity) {
+
+				// while zooming in/out, scale both unit name labels, attribute bars, and chatBubble
+				if (ige.pixi.viewport.isZooming) {
```

Comment:
yeah this is the one I was getting hung up on. I think I may have a solution, but I think it would have to involve adding a boolean prop to `ige.client`. Same way viewport is doing it, we instead check `if (ige.client.isZooming)`

Reply:
also looking at these now, they are all updates to scale and position, and I think they don't belong in `updateAllEntities`